### ISSUE TITLE
Somoclu is unable to find matplotlib

### ISSUE THREAD

**Issue by rohitgavval** [peterwittek/somoclu#101]
I am trying to install somoclu cloned yesterday from github. 

Environment:
Ubuntu 14.04 
Python 2.7.6 
CUDA 7.5

I am running this inside a python virutal environment. 

The preparatory steps went fine but when I run the setup.py, the installation is unable to find matplotlib and it is failing. I tried installing matplotlib before running the setup file but still face the same issue. I checked that the proxy is properly setup and also set it up in the .bashrc of the virtual environment to be double sure. Below is the tail of the log. 

`Extracting somoclu-1.7.4-py2.7.egg to /usr/local/lib/python2.7/dist-packages
Removing somoclu 1.7.4 from easy-install.pth file
Adding somoclu 1.7.4 to easy-install.pth file

Installed /usr/local/lib/python2.7/dist-packages/somoclu-1.7.4-py2.7.egg
Processing dependencies for somoclu==1.7.4
Searching for matplotlib
Reading https://pypi.python.org/simple/matplotlib/
Download error on https://pypi.python.org/simple/matplotlib/: [Errno 101] Network is unreachable -- Some packages may not be found!
Couldn't find index page for 'matplotlib' (maybe misspelled?)
Scanning index of all packages (this may take a while)
Reading https://pypi.python.org/simple/
Download error on https://pypi.python.org/simple/: [Errno 101] Network is unreachable -- Some packages may not be found!
No local packages or download links found for matplotlib
error: Could not find suitable distribution for Requirement.parse('matplotlib')
`

Any pointers on how to fix this?

**Comment by xgdgsc** [comment]
Changing pypi mirror?

**Comment by rohitgavval** [comment]
I can't seem to find the location to change the pypi mirror. Is it something defined in the somoclu files or the default python installation?

**Comment by xgdgsc** [comment]
like https://mirrors.tuna.tsinghua.edu.cn/help/pypi/

**Comment by rohitgavval** [comment]
Yes, but could you tell me how to tell the installer which pypi mirror to use?

**Comment by xgdgsc** [comment]
https://coderwall.com/p/lykptw/tell-pip-to-use-a-mirror
